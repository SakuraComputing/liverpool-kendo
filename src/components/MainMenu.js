import React from 'react';
import { slide as Menu } from 'react-burger-menu'
import { link } from 'react-router';

class MainMenu extends React.Component {

    showSettings (event) {
        event.preventDefault();
        alert('show settings');
    }

    render () {
        return (
            <Menu className="main-menu" width={ '50px' }>
                <a id="home" className="menu-item" href="/">Home</a>
                <a id="about" to="/about" className="menu-item" href="/about">About</a>
                <a id="contact" className="menu-item" href="/contact">Contact</a>
                <a onClick={ this.showSettings } className="menu-item--small" href="">Settings</a>
            </Menu>
        );
    }
}
export default MainMenu
