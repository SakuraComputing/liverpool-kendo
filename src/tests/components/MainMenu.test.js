import React from 'react';
import { shallow } from 'enzyme';
import MainMenu from '../../components/MainMenu';

describe('Maine Menu', () => {
    it('should render the form correctly', () => {
        const wrapper = shallow(<MainMenu/>);
        expect(wrapper).toMatchSnapshot();
    });
});