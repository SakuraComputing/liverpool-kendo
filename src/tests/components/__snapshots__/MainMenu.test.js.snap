// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maine Menu should render the form correctly 1`] = `
<Menu
  bodyClassName=""
  burgerBarClassName=""
  burgerButtonClassName=""
  className="main-menu"
  crossButtonClassName=""
  crossClassName=""
  id=""
  itemListClassName=""
  menuClassName=""
  morphShapeClassName=""
  noOverlay={false}
  onStateChange={[Function]}
  outerContainerId=""
  overlayClassName=""
  pageWrapId=""
  styles={Object {}}
  width="50px"
>
  <a
    className="menu-item"
    href="/"
    id="home"
  >
    Home
  </a>
  <a
    className="menu-item"
    href="/about"
    id="about"
    to="/about"
  >
    About
  </a>
  <a
    className="menu-item"
    href="/contact"
    id="contact"
  >
    Contact
  </a>
  <a
    className="menu-item--small"
    href=""
    onClick={[Function]}
  >
    Settings
  </a>
</Menu>
`;
